### DISCUSSION TITLE
Question from a newcomer about handling multigames pgn's

### DISCUSSION THREAD

**Discussion by Huub62** [mliebelt/pgn-viewer#562] [Q&A]
I am new to using this script so please forgive me if i ask something thats obvious for many of you.

I made a custom layout for a pgn viewer and all worked really well. But i ran into a problem when i tried to upload a pgn file with multiple games in it.

First i checked the example on: https://mliebelt.github.io/pgn-viewer/examples/1101.html

Then worked on some code to upload the pgn file and process it correctly. But when i tested it, the dropdownlist was indeed filled and the first game could be played, but as soon as i choose another from the dropdownlist this error showed up: 

```
dist.js:1 Uncaught TypeError: Cannot read properties of null (reading 'parentNode')
    at i (dist.js:1:951316)
    at HTMLSelectElement.<anonymous> (dist.js:1:950460)
```
   
Of course i checked the pgn file but it pasted the test in the Lichess editor and i also tried some very simple pgn files, but the error remained. When i looked back at the website: https://mliebelt.github.io/pgn-viewer/examples/1101.html and opened the console i noticed it showed the same error. Can someone please help me solve this issue. What i hope to accomplish is to have a uploadfile button on my page that gets the dropdownlist filled with the games and that they can be played back. As it is now i not only get the error but also the board doesnt update to the new game choosen from the dropdownlist. So, right now it is useless. My current code is:

 

    
```
            document.getElementById('pgnFile').addEventListener('change', function(event) {
            const file = event.target.files[0];
            if (file) {
                const reader = new FileReader();
                reader.onload = function(e) {
                    const pgnText = e.target.result;    
                    loadPgn(pgnText);
                };
                reader.readAsText(file);
            }
        });
    
        function loadPgn(pgnText) {
            pgnv = PGNV.pgnView("board", {
                pgn: pgnText,
                manyGames: true,
                showCoords: true, 
                orientation: 'white', 
                theme: 'brown', 
                pieceStyle: 'merida',
                figurine: 'merida', 
                locale: 'nl', 
                width: '770px', 
                boardsize: '400px', 
                layout: 'right', 
                timerTime: 0, 
                coordsInner: true, 
                coordsFactor: 1.0,
                coordsFontSize: 14, 
                notation: 'short', 
                notationLayout: 'list', 
                hideMovesBefore: false, 
                showFen: false, 
                colorMarker: 'cm-big'
            });
        }
    
        window.onload = function() {
            loadPgn('');
        };

```
Thank you all in advance for any help and wishing you all a wonderful day!

**Comment by mliebelt** [comment]
Your idea sounds plausible, have to check that. I have not touched the reader/viewer for months, so I am a little bit rusty, and the code is not so easy to understand. I have to check your example ...  So by having the reference to the example code, it should be relatively straight forward to see why that is the case.

**Reply by Huub62** [reply]
Any help would be greatly appreciated :) It felt like such a normal thing to have an upload for the pgn games. It would for instance allow trainers, teacher to have a folder, google drive, usb-stick with there pgn-files for there chesslessons and use then anywhere. They all can maintain there own pgn-files or get send pgn-files and all use the same tool to view them and use them in there lessons.

**Reply by mliebelt** [reply]
I created a ticket from your post (https://github.com/mliebelt/pgn-viewer/issues/563), and hopefully found the reason for it. It was there all the time, never worked (well), but your post triggered me to search for the reason. Not sure about it, but it is worth doing a new release for it. Will perhaps take a little time (hopefully not more than tomorrow), before the release is there.

**Comment by mliebelt** [comment]
Went smooth yesterday, so the release is out. Please check, if that solves your problem. @Huub62

**Reply by Huub62** [reply]
I send you a reply to your email earlier today.

The error is gone but it didnt solve the issues with the multigames pgn.

If i now upload a simple pgn with only complete games and nothing special in it, for instance the twic-01-232games.pgn then all works just fine. But if i try a more complicated  pgn, with positions, variations, comments etc. then it crashes. The pgn's I tried it with I have tested on chessbase, chess.com and lichess and never cause any problems, so I must assume they are correct.

Here is what happens ...
If i upload the file it correctly fills the dropdown list and the start position from the first game is shown. If however I start to pick other games/positions from the list two things can happen  .....the board no longer updates and/or the error "Uncaught Error: No legal move" appears.

Once the error has occurred nothing works anymore. No boards update and every click on every move causes another error. The error even mentions moves that aren't related to the position on the board, but are from other games.

In the email i send you i attached the pgn and the html file I use.

**Reply by mliebelt** [reply]
I don't have an email from you, and it is the wrong way to do that. Please file a new ticket, have there a (minimal) PGN file to reproduce the error included, and then I will have a look. This stream here is for discussions, and gets messy. I have an idea where the problem comes from, but I have to check (first). And having a minimal example would be good. I have used in the past the multigame part alone with complete games from the start position, so I think that is the reason why it does not work out of the box. And the error `No legal move` is a result from that.
